app/src/main/java/com/example/myapplication/ui/professorsList/adapter/ProfessorAdapter.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.myapplication.ui.professorsList.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import com.example.common.entity.Professor
+import com.example.myapplication.R
+import com.example.myapplication.ui.professorsList.viewholder.ProfessorViewHolder
+
+class ProfessorAdapter(
+    callback: DiffUtil.ItemCallback<Professor>,
+    private val onClickListener: OnClickListener
+) : ListAdapter<Professor, ProfessorViewHolder>(callback) {
+
+    interface OnClickListener {
+        fun onClick(position: Int)
```

Comment:
`onClick` should take in the actual Professor item as its parameter not its position. Since the view cannot retrieve the Professor list from a ListAdapter. And even if it could (e.g by using a basic RecyclerView.Adapter), it would create unnecessary coupling between view (fragment) and adapter.